HULKs/hulk
# Tooling version checks

## Introduced Changes

Adds checks on startup to pepsi and twix which compare the compiled version number to the one found in their Cargo.toml in the repository.
If the running version is lower than the one in the repository a message saying so is shown to the user.
A copy-pastable `cargo install` command is also shown to install the newer version.

We regularly had issues with people running an outdated version of pepsi, which would then incorrectly build the code or miss files/configuration during upload.
Properly versioning should solve this, however, it does require us to keep bumping the version number when making changes.

Fixes #404

## ToDo / Known Issues

## Ideas for Next Iterations (Not This PR)

- Indicate that there is an update available in the twix UI.
- Allow installing tools via pepsi. `pepsi install {pepsi,twix}`?

## How to Test

Since these changes only take effect once the current version is outdated, you need to first `cargo install` the tools, then bump the version number in `tools/{pepsi,twix}/Cargo.toml`, then run the tools to see the message.
